src/site/apt/sessions/session-access-grants.apt.vm

```diff
@@ -3,6 +3,123 @@ Access Grants
+Request req = Request.newBuilder(sharedTextFileURI)
+        .HEAD()
+        .build();
+Response res = authClient.send(req, Response.BodyHandlers.discarding());
+res.headers().allValues("Link").stream()
+    .flatMap(l -> Headers.Link.parse(l).stream())
+    .forEach(link -> {
+        if (link.getParameter("rel").contains("acl")) {
+            acp.add(link.getUri());
+        }
+    });
+if (!acp.isEmpty()) {
+    URI oneAcp = acp.get(0);
+    IRI acpResource = rdf.createIRI(oneAcp.toString());
+
+    Dataset dataset = rdf.createDataset();
+
+    //creating a new matcher
+    URI newMatcherURI = URIBuilder.newBuilder(oneAcp).fragment("newMatcher").build();
+    IRI newMatcher = rdf.createIRI(newMatcherURI.toString());
+    IRI rdfType = rdf.createIRI(com.inrupt.client.vocabulary.RDF.type.toString());
+    IRI matcherType = rdf.createIRI("http://www.w3.org/ns/solid/acp#Matcher");
+    dataset.add(rdf.createQuad(acpResource, newMatcher, rdfType, matcherType));
+
+    IRI vc = rdf.createIRI("http://www.w3.org/ns/solid/acp#vc");
+    IRI solidAccessGrant = rdf.createIRI("http://www.w3.org/ns/solid/vc#SolidAccessGrant");
+    dataset.add(rdf.createQuad(acpResource, newMatcher, vc, solidAccessGrant));
+
+    //create a new policy
+    URI newPolicyURI = URIBuilder.newBuilder(oneAcp).fragment("newPolicy").build();
+    IRI newPolicy = rdf.createIRI(newPolicyURI.toString());
+    IRI policyType = rdf.createIRI("http://www.w3.org/ns/solid/acp#Policy");
+    dataset.add(rdf.createQuad(acpResource, newPolicy, rdfType, policyType));
+
+    IRI allOf = rdf.createIRI("http://www.w3.org/ns/solid/acp#allOf");
+    dataset.add(rdf.createQuad(acpResource, newPolicy, allOf, newMatcher));
+
+    IRI allow = rdf.createIRI("http://www.w3.org/ns/solid/acp#allow");
+    IRI read = rdf.createIRI("http://www.w3.org/ns/solid/acp#Read");
+    dataset.add(rdf.createQuad(acpResource, newPolicy, allow, read));
+
+    IRI write = rdf.createIRI("http://www.w3.org/ns/solid/acp#Write");
+    dataset.add(rdf.createQuad(acpResource, newPolicy, allow, write));
+    
+    //creating a new access control
+    URI newAccessControlURI = URIBuilder.newBuilder(oneAcp).fragment("newAccessControl").build();
+    IRI newAccessControl = rdf.createIRI(newAccessControlURI.toString());
+    IRI accessControlType = rdf.createIRI("http://www.w3.org/ns/solid/acp#AccessControl");
+    dataset.add(rdf.createQuad(acpResource, newAccessControl, rdfType, accessControlType));
+
+    IRI apply = rdf.createIRI("http://www.w3.org/ns/solid/acp#apply");
+    dataset.add(rdf.createQuad(acpResource, newAccessControl, apply, newPolicy));
+
+    //adding the new access control to the ACP
+    IRI accessControl = rdf.createIRI("http://www.w3.org/ns/solid/acp#accessControl");
+    dataset.add(rdf.createQuad(acpResource, newAccessControl, accessControl, newAccessControl));
+
+    try (SolidRDFSource updatedAcp = new SolidRDFSource(oneAcp, dataset, null)) {
+        authClient.update(updatedAcp));
+    }
+}
```

Comment:
As discussed, maybe:
1. extract the vocabularies,
2. remove the unnecessary types (all but `acp:AccessControlResource`).